### PR TITLE
TASK: Migrate to Neos 3.0

### PR DESCRIPTION
**Pull request by simonschaufi**
Releases: master
Relates: https://github.com/SimplyAdmire/SimplyAdmire.Neos.FormBuilderBundle/pull/10

### CODE REVIEW

**Review by kdambekalns — APPROVED** [review]
Looks good. What about making the package use PSR-4 while it is being updated anyway?

**Comment by simonschaufi** [comment]
Sure, I can do that. Could you make me a short checklist what needs to be done please?

**Comment by radmiraal** [comment]
@simonschaufi change the autoload config in composer.json from:

```
		"psr-0": {
			"TYPO3\\FormBuilder": "Classes"
		}
```
to (mind the trailing slashes)
```
		"psr-4": {
			"Neos\\FormBuilder\\": "Classes"
		}
```

and then move all files from Classes/Neos/FormBuilder/ directly into the Classes folder

**Comment by simonschaufi** [comment]
I changed super much and had to force push (don't know why). Can you review please?

**Comment by simonschaufi** [comment]
Can somebody also take care about the new composer name? I don't know if anything needs to be done on packagist

File: Classes/Controller/EditorController.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
+<?php
+namespace Neos\FormBuilder\Controller;
+
+/*                                                                        *
+ * This script belongs to the TYPO3 Flow package "Neos.FormBuilder".      *
```
Comment (htuscher) [inline Classes/Controller/EditorController.php]: Please use new header (no TYPO3)

File: Classes/Controller/FormManagerController.php [hunk 2]
```diff
@@ -0,0 +1,165 @@
+<?php
+namespace Neos\FormBuilder\Controller;
+
+/*                                                                        *
+ * This script belongs to the TYPO3 Flow package "Neos.FormBuilder".      *
```
Comment (htuscher) [inline Classes/Controller/FormManagerController.php]: Please use new file header (no TYPO3)

File: Classes/Controller/StandardController.php [hunk 3]
```diff
@@ -1,8 +1,8 @@
 <?php
-namespace TYPO3\FormBuilder\Controller;
+namespace Neos\FormBuilder\Controller;
 
 /*                                                                        *
- * This script belongs to the TYPO3 Flow package "TYPO3.FormBuilder".     *
+ * This script belongs to the TYPO3 Flow package "Neos.FormBuilder".      *
```
Comment (htuscher) [inline Classes/Controller/StandardController.php]: Please use new file header (no TYPO3)

File: Classes/ViewHelpers/JsonConfigurationViewHelper.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
+<?php
+namespace Neos\FormBuilder\ViewHelpers;
+
+/*                                                                        *
+ * This script belongs to the TYPO3 Flow package "Neos.FormBuilder".      *
```
Comment (htuscher) [inline Classes/ViewHelpers/JsonConfigurationViewHelper.php]: Please use new file header (no TYPO3)

File: README.md [hunk 5]
```diff
@@ -75,7 +75,7 @@ Each handlebars template which is used by the Form Builder is registered in the
 	TYPO3:
```
Comment (htuscher) [inline README.md]: Please Change Namespace to Neos

File: Resources/Private/CoffeeScript/app.coffee [hunk 6]
```diff
@@ -11,14 +11,14 @@
 
 # #Main Base Class#
 # Main Entry point for the form builder, setting up the environment and initializing
-# `TYPO3.FormBuilder` namespace.
+# `Neos.FormBuilder` namespace.
 #
 
 TYPO3 = window.TYPO3 || {}
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/app.coffee]: Please change to Neos

File: Resources/Private/CoffeeScript/utility.coffee [hunk 7]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the TYPO3 Flow package "TYPO3.FormBuilder".
+# This script belongs to the TYPO3 Flow package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/utility.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/CoffeeScript/view.coffee [hunk 8]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the TYPO3 Flow package "TYPO3.FormBuilder".
+# This script belongs to the TYPO3 Flow package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/CoffeeScript/view/application.coffee [hunk 9]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the FLOW3 package "TYPO3.FormBuilder".
+# This script belongs to the FLOW3 package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view/application.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/CoffeeScript/view/elementOptionsPanel.coffee [hunk 10]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the FLOW3 package "TYPO3.FormBuilder".
+# This script belongs to the FLOW3 package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view/elementOptionsPanel.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/CoffeeScript/view/elementOptionsPanelEditors/basic.coffee [hunk 11]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the FLOW3 package "TYPO3.FormBuilder".
+# This script belongs to the FLOW3 package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view/elementOptionsPanelEditors/basic.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/CoffeeScript/view/elementOptionsPanelEditors/collectionEditor.coffee [hunk 12]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the FLOW3 package "TYPO3.FormBuilder".
+# This script belongs to the FLOW3 package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view/elementOptionsPanelEditors/collectionEditor.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/CoffeeScript/view/elementOptionsPanelEditors/finisher.coffee [hunk 13]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the FLOW3 package "TYPO3.FormBuilder".
+# This script belongs to the FLOW3 package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view/elementOptionsPanelEditors/finisher.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/CoffeeScript/view/elementOptionsPanelEditors/grid.coffee [hunk 14]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the FLOW3 package "TYPO3.FormBuilder".
+# This script belongs to the FLOW3 package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view/elementOptionsPanelEditors/grid.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/CoffeeScript/view/elementOptionsPanelEditors/validator.coffee [hunk 15]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the FLOW3 package "TYPO3.FormBuilder".
+# This script belongs to the FLOW3 package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view/elementOptionsPanelEditors/validator.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/CoffeeScript/view/elementOptionsPanelEditors/validator.coffee [hunk 16]
```diff
@@ -88,7 +88,7 @@ TYPO3.FormBuilder.View.ElementOptionsPanel.Editor.ValidatorEditor = TYPO3.FormBu
 # methods to be used by subclasses and does not render any editing options.
 #
 # ###Public API
-TYPO3.FormBuilder.View.ElementOptionsPanel.Editor.ValidatorEditor.DefaultValidatorEditor = Ember.View.extend {
+Neos.FormBuilder.View.ElementOptionsPanel.Editor.ValidatorEditor.DefaultValidatorEditor = Ember.View.extend {
 	classNames: ['typo3-formbuilder-validator-editor']
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view/elementOptionsPanelEditors/validator.coffee]: Please rename class without typo3

File: Resources/Private/CoffeeScript/view/header.coffee [hunk 17]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the FLOW3 package "TYPO3.FormBuilder".
+# This script belongs to the FLOW3 package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view/header.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/CoffeeScript/view/insertElementsPanel.coffee [hunk 18]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the FLOW3 package "TYPO3.FormBuilder".
+# This script belongs to the FLOW3 package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view/insertElementsPanel.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/CoffeeScript/view/stage.coffee [hunk 19]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the FLOW3 package "TYPO3.FormBuilder".
+# This script belongs to the FLOW3 package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view/stage.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/CoffeeScript/view/structurePanel.coffee [hunk 20]
```diff
@@ -1,5 +1,5 @@
 # <!--
-# This script belongs to the FLOW3 package "TYPO3.FormBuilder".
+# This script belongs to the FLOW3 package "Neos.FormBuilder".
```
Comment (htuscher) [inline Resources/Private/CoffeeScript/view/structurePanel.coffee]: Please use new file header (no TYPO3)

File: Resources/Private/FormBuilderTemplates/Application.html [hunk 21]
```diff
@@ -1,20 +1,20 @@
-{{view TYPO3.FormBuilder.View.Header id="typo3-formbuilder-header" tagName="header"}}
+{{view Neos.FormBuilder.View.Header id="typo3-formbuilder-header" tagName="header"}}
 
-{{view TYPO3.FormBuilder.View.Stage tagName="section" id="typo3-formbuilder-stage"}}
+{{view Neos.FormBuilder.View.Stage tagName="section" id="typo3-formbuilder-stage"}}
 
 <section id="typo3-formbuilder-elementSidebar">
```
Comment (htuscher) [inline Resources/Private/FormBuilderTemplates/Application.html]: please remove typo3 from ids

File: Resources/Private/FormBuilderTemplates/ElementOptionsPanel/FinisherEditor.html [hunk 22]
```diff
@@ -1,5 +1,5 @@
 <h3>Finishers</h3>
 <div class="typo3-formbuilder-addFinisher">
```
Comment (htuscher) [inline Resources/Private/FormBuilderTemplates/ElementOptionsPanel/FinisherEditor.html]: please remove typo3 from classname

File: Resources/Private/FormBuilderTemplates/ElementOptionsPanel/IdentifierEditor.html [hunk 23]
```diff
@@ -2,7 +2,7 @@
 	<label>Identifier:</label>
 	<div class="typo3-formbuilder-controls">
```
Comment (htuscher) [inline Resources/Private/FormBuilderTemplates/ElementOptionsPanel/IdentifierEditor.html]: please remove typo3 from classname

File: Resources/Private/FormBuilderTemplates/ElementOptionsPanel/TextEditor.html [hunk 24]
```diff
@@ -1,6 +1,6 @@
 <div class="typo3-formbuilder-controlGroup">
```
Comment (htuscher) [inline Resources/Private/FormBuilderTemplates/ElementOptionsPanel/TextEditor.html]: please remove typo3 from classname

File: Resources/Private/FormBuilderTemplates/ElementOptionsPanel/TextEditor.html [hunk 25]
```diff
@@ -1,6 +1,6 @@
...
 	<label>{{label}}:</label>
 	<div class="typo3-formbuilder-controls">
```
Comment (htuscher) [inline Resources/Private/FormBuilderTemplates/ElementOptionsPanel/TextEditor.html]: please remove typo3 from classname

File: Resources/Private/FormBuilderTemplates/ElementOptionsPanel/ValidatorEditor.html [hunk 26]
```diff
@@ -1,5 +1,5 @@
 <h3>Validators</h3>
 <div class="typo3-formbuilder-addValidator">
```
Comment (htuscher) [inline Resources/Private/FormBuilderTemplates/ElementOptionsPanel/ValidatorEditor.html]: please remove typo3 from classname

File: Resources/Private/FormBuilderTemplates/Header.html [hunk 27]
```diff
@@ -1,14 +1,14 @@
-<h1>Form Builder - {{TYPO3.FormBuilder.Model.Form.formDefinition.label}}</h1>
-{{#if TYPO3.FormBuilder.Model.Form.currentlyLoadingPreview}}<span id="typo3-formbuilder-loading">Loading..</span>{{/if}}
+<h1>Form Builder - {{Neos.FormBuilder.Model.Form.formDefinition.label}}</h1>
+{{#if Neos.FormBuilder.Model.Form.currentlyLoadingPreview}}<span id="typo3-formbuilder-loading">Loading..</span>{{/if}}
 
 <ul id="typo3-formbuilder-toolbar">
```
Comment (htuscher) [inline Resources/Private/FormBuilderTemplates/Header.html]: please remove typo3 from classname

File: Resources/Private/FormBuilderTemplates/InsertElementsPanel.html [hunk 28]
```diff
@@ -1,4 +1,4 @@
 {{#collection contentBinding="formElementsGrouped"}}
 	<h3 class="typo3-formbuilder-group-{{unbound content.key}}"><span>{{content.label}}</span></h3>
```
Comment (htuscher) [inline Resources/Private/FormBuilderTemplates/InsertElementsPanel.html]: please remove typo3 from classname

File: Resources/Private/FormBuilderTemplates/Validator/MinimumMaximumEditor.html [hunk 29]
```diff
@@ -3,13 +3,13 @@
 <div class="typo3-formbuilder-controlGroup">
 	<label>Minimum:</label>
 	<div class="typo3-formbuilder-controls">
```
Comment (htuscher) [inline Resources/Private/FormBuilderTemplates/Validator/MinimumMaximumEditor.html]: please remove typo3 from classname

**Review by htuscher — CHANGES_REQUESTED** [review]
Please remove all traces of typo3

**Comment by simonschaufi** [comment]
If you want to, you can actually make changes in my code. Otherwise you need to wait some more time until i have done every change you requested.

**Comment by htuscher** [comment]
I created a PR based on your changes to remove all traces of typo3 https://github.com/simonschaufi/form-builder/pull/1

**Comment by simonschaufi** [comment]
thank you very much!

**Review by htuscher — APPROVED** [review]

**Comment by simonschaufi** [comment]
can this be merged?

**Comment by albe** [comment]
Let's do this :) Thanks for the effort!

**Comment by simonschaufi** [comment]
on packagist the composer name is still typo3/formbuilder but it should be neos/formbuilder. what needs to be done?

**Comment by albe** [comment]
I guess someone with access on packagist needs to rename/move the package. @kdambekalns maybe?

**Comment by kdambekalns** [comment]
Done.

**Pull request merged**